protocols/migration/migration_reports_5_0_0_to_reports_6_0_0.py

```diff
@@ -25,3 +25,172 @@ def migrate_interpretation_request_rd(self, old_instance):
         return self.validate_object(
             object_to_validate=new_instance, object_type=self.new_model.InterpretationRequestRD
         )
+
+    def migrate_interpreted_genome_rd(self, old_instance):
+        """
+        Migrates a reports_5_0_0.InterpretedGenomeRD into a reports_6_0_0.InterpretedGenome
+        :type old_instance: reports_5_0_0.InterpretedGenomeRD
+        :rtype: reports_6_0_0.InterpretedGenome
+        """
+        new_instance = self.convert_class(self.new_model.InterpretedGenome, old_instance)
+
+        new_instance.variants = self.migrate_variants(old_variants=old_instance.variants)
+
+        return self.validate_object(
+            object_to_validate=new_instance, object_type=self.new_model.InterpretedGenome
+        )
+
+    def migrate_variants(self, old_variants):
+        return [self.migrate_variant(old_variant=old_variant) for old_variant in old_variants]
+
+    def migrate_variant(self, old_variant):
+        """
+        Migrates a reports_5_0_0.ReportedVariant to a reports_6_0_0.SmallVariant
+        :param old_variant: reports_5_0_0.ReportedVariant
+        :return: reports_6_0_0.SmallVariant
+        """
+
+        new_variant = self.convert_class(self.new_model.SmallVariant, old_variant)
+        new_variant.variantCoordinates = self.convert_class(self.new_model.VariantCoordinates, old_variant.variantCoordinates)
+        new_variant.variantCalls = self.migrate_variant_calls(variant_calls=old_variant.variantCalls)
+        new_variant.reportEvents = self.migrate_report_events(report_events=old_variant.reportEvents)
+        new_variant.variantAttributes = self.migrate_variant_attributes(old_variant=old_variant)
+        return self.validate_object(
+            object_to_validate=new_variant, object_type=self.new_model.SmallVariant
+        )
+
+    def migrate_variant_attributes(self, old_variant):
+        # This logic is done here because alleleOrigins is part of the
+        # v6 variantAttributes, which is non-nullable in v5 (or v6)
+        # so the v5 variantAttributes could be null, but the alleleOrigins can not be,
+        # so this is done to make sure we don't lose the alleleOrigins
+        if old_variant.variantAttributes is not None:
+            new_variant_attributes = self.convert_class(self.new_model.VariantAttributes, old_variant.variantAttributes)
+        else:
+            new_variant_attributes = self.new_model.VariantAttributes()
+
+        new_variant_attributes.genomicChanges = old_variant.genomicChanges
```

Comment:
why do we need to map these fields? I thought these would be filled by `self.convert_class()` as they are named the same.
These comment applies to lines from 72 to 78, both inclusive.

Reply:
I understand now why it is needed